### PR TITLE
Implement pieces of planned Commands API in support of AI App Generator card

### PR DESCRIPTION
**Pull request by lukemelia**
This PR does not include the finished vision for the Commands API, but it is a helpful merge point.

Still to-do in future PRS:

* get module creation working smoothly via commands
* sample data generation via commands
* Use loader to access host commands via import instead of `loadCommand` API
* Add tools to chat as roomState similar to how we treat Skills and attached Cards
* Update system prompt to make sure ai knows which tools are auto-executable and which are not

### CODE REVIEW

File: packages/ai-bot/lib/matrix.ts [hunk 1]
```diff
@@ -126,7 +126,13 @@ export const toMatrixMessageCommandContent = (
   eventToUpdate: string | undefined,
 ): IContent | undefined => {
   let { arguments: payload } = functionCall;
-  const body = payload['description'] || 'Issuing command';
+  // Take the description if it exists (old format)
+  // Otherwise take the description from the attributes (new format, responses are cards)
+  // If neither exists, use a default
+  const body =
+    payload['description'] ||
+    payload['attributes']?.description ||
```
Comment (lukemelia) [inline packages/ai-bot/lib/matrix.ts]: @IanCal I don't think we should ever use this one. What prompted you to add it?

File: packages/ai-bot/main.ts [hunk 2]
```diff
@@ -24,9 +24,15 @@ import { handleDebugCommands } from './lib/debug';
 import { MatrixClient } from './lib/matrix';
 import type { MatrixEvent as DiscreteMatrixEvent } from 'https://cardstack.com/base/matrix-event';
 import * as Sentry from '@sentry/node';
+import fs from 'fs';
 
 let log = logger('ai-bot');
 
+// Ensure logs directory exists
+if (!fs.existsSync('ai-bot-logs')) {
+  fs.mkdirSync('ai-bot-logs');
```
Comment (lukemelia) [inline packages/ai-bot/main.ts]: @IanCal do we want to remove this?

Comment (lukemelia) [reply packages/ai-bot/main.ts]: Or flag it?

Comment (IanCal) [reply packages/ai-bot/main.ts]: yes I'll clear that all out

File: packages/ai-bot/lib/matrix.ts (hunk 1 above)
Comment (IanCal) [reply packages/ai-bot/lib/matrix.ts]: if I'm picturing the layering correctly this is what we currently do for the patch command (it takes a description as an input which is what we show in the message text) but I'd be happy to remove it.

File: packages/base/command.gts [hunk 3]
```diff
@@ -49,3 +51,41 @@ export class CommandCard extends CardDef {
...
+}
+
+export class PatchCardInput extends CardDef {
+  @field cardId = contains(StringField);
+  @field patch = contains(JsonField); //TODO: JSONField ?
+}
+
+export class ShowCardInput extends CardDef {
+  @field cardToShow = linksTo(CardDef);
+  @field placement = contains(StringField); // TODO: nicer if enum, likely need to specify stackIndex too?
```
Comment (habdelra) [inline packages/base/command.gts]: take a look at base64-image's `size` field for an example of enum field

File: packages/catalog-realm/AiAppGenerator/commands.ts
Comment (habdelra) [inline packages/catalog-realm/AiAppGenerator/commands.ts:1]: there's a lot of different classes in this single module. I can see this growing into a dumping ground for commands. perhaps a more extensible way to treat this is to make a folder like: `catalog-realm/AiAppGenerator/commands`. and have a separate module for each command, and an `index.ts` that reexports all the commands in the folder.

File: packages/catalog-realm/ai-app-generator.gts [hunk 4]
```diff
@@ -347,66 +351,85 @@ class DashboardTab extends GlimmerComponent<{
...
+  //         );
+  //         this.prompt = this.promptReset;
+  //         this.args.setActiveTab?.('requirements');
+  //       } catch (e) {
+  //         this.errorMessage =
+  //           e instanceof Error ? `${e.name}: ${e.message}` : 'An error occurred.';
+  //         throw e;
+  //       }
+  //   },
+  // );
```
Comment (habdelra) [inline packages/catalog-realm/ai-app-generator.gts]: clean this up please

File: packages/catalog-realm/product-requirement-document.gts [hunk 5]
```diff
@@ -13,8 +13,7 @@ import {
 import { Button } from '@cardstack/boxel-ui/components';
 import { ImagePlaceholder } from '@cardstack/boxel-ui/icons';
 import { bool, cn, not } from '@cardstack/boxel-ui/helpers';
-import { on } from '@ember/modifier';
-import { action } from '@ember/object';
+import { on } from '@ember/modifier'; // TODO: why is this a type error???
```
Comment (habdelra) [inline packages/catalog-realm/product-requirement-document.gts:16]: hmmm, I vaguely recall doing something special fro base realm and experiments realm to get the types correct, we need to do that here too

Comment (habdelra) [reply packages/catalog-realm/product-requirement-document.gts:16]: look at packages/base/types/index.d.ts

File: packages/catalog-realm/product-requirement-document.gts [hunk 6]
```diff
@@ -248,12 +247,12 @@ class Isolated extends Component<typeof ProductRequirementDocument> {
     return this.args.model[realmURL];
   }
 
-  @action generateApp() {
+  generateApp = () => {
```
Comment (habdelra) [inline packages/catalog-realm/product-requirement-document.gts:250]: lol, i prefer this form better too. seems less magical....

File: packages/base/command.gts (hunk 3 above)
Comment (lukemelia) [reply packages/base/command.gts]: Thanks! Removing this field for now. Will revisit

**Review by habdelra — APPROVED** [review]
some minor items, but looks fine overall

**Pull request merged**